packages/window-size-tool/rules.json

```diff
@@ -3,6 +3,9 @@
     "width": "< 1200",
     "terms": ["//CLUSTER/@NUM_COLS > 4", "count(//LIST[count(FIELD | CONTAINER) > 6 ]) > 0"
     ],
+    "andTerms": [
+      "count(//INCLUDE) = 0 and count(//MENU[@MODE='WIZARD_PROGRESS_BAR']) = 0" 
```

Comment:
`andTerms` is an array that gets looped over in your engine, requiring each terms to be true, so you could write each of these expressions as separate items in the array and, I think, they'd be easier to read and write.

e.g.
```
"andTerms": [
      "count(//INCLUDE) = 0",
      "count(//MENU[@MODE='WIZARD_PROGRESS_BAR']) = 0"
     ],
```

Reply:
It looks like this change will not work Graham, The logic in the engine checks that if any of the` anyTerms` is evaluated to true, then it loops over the  `allTerms` to check if that's also true, If the ` `allTerms`` is also evaluated to true then the rule is applied. 
Therefore I think it should be as it was
```
"andTerms": [
      "count(//INCLUDE) = 0 and count(//MENU[@MODE='WIZARD_PROGRESS_BAR']) = 0" 
],
```

Reply:
> The logic in the engine checks that if any of the anyTerms is evaluated to true, then it loops over the allTerms to check if that's also true, If the allTerms is also evaluated to true then the rule is applied.

Correct, as I'd expect.

but the point of making `allTerms` an array was so that we could make writing multiple terms easier, so instead of requiring an `and` operator you can just write each as an item in an array, otherwise why make `allTerms` an array?

maybe i'm reading it wrong but it looks like to me the new loop yo've written supports this no?

Reply:
Yes, that’s what I would expect, because if you look at the code, the looping part is similar/same as what we have for the first part.

So the output would have been the same wether it's an `and` or not!  I could have a look into that in more details but as this is a temporary solution and we're already getting the result we expect, would this worth spending more time on it?

Reply:
Although we intend that ignoring VIM and Wizards _might_ to be temporary or it might not be.

Enabling the ability to specify a set of terms that must _all_ be met is a really useful feature, not temporary and worth getting right. We could actually make a number of ouor other rules a lot easier to read and write by using this feature.